end2end/tests/reportBuilder.spec.ts

```diff
@@ -2,30 +2,208 @@ import { test, expect } from '@playwright/test';
+    await page.getByText('General Workflow - Group').click();
+    await page.getByText('General Workflow - Requestor').click();
+    await page.getByRole('button', { name: 'Generate Report' }).click();
+    await expect(page.getByRole('cell', { name: 'No Results' })).toBeVisible();
+});
+
+test("Take Action allow user to perform necessary form Action", async ({ page }) => {
+    await page.goto("https://host.docker.internal/Test_Request_Portal/?a=reports&v=3&query=N4IgLgpgTgtgziAXAbVASwCZJHSAHASQBEQAaEAez2gEMwKpsBCAXjJBjoGMALbKCHAoAbAG4Qs5AOZ0I2AIIA5EgF9S6LIhAYIwiJEmVqUOg2xtynMLyQAGabIXKQKgLrkAVhTQA7BChwwOgBXBHccBjAkYDUQYTQYNCjEAEZbdPJ4xLAAeQAzPLh9OxUgA&indicators=NobwRAlgdgJhDGBDALgewE4EkAiYBcYyEyANgKZgA0YUiAthQVWAM4bL4AMAvpeNHCRosuAi2QoAri2a0G%2BMMzboOeHn0iwEKDDgWJ4RVFABCk5Giiz6jRdWWqeAXSA%3D");
+    await page.getByRole('row', { name: '955 Take Action Available for' }).locator('div').click();
+    await page.getByLabel('comment text area').click();
+    await page.getByLabel('comment text area').fill('testing purpose');
+    await page.getByRole('button', { name: 'Close' }).click();
+
+    // un-comment if need to Approve Action 
+    // await page.getByRole('button', { name: 'Approve' }).click();
+});
+
+test("Test Share Report button", async ({ page }) => {
+    await page.goto("https://host.docker.internal/Test_Request_Portal/?a=reports&v=3&query=N4IgLgpgTgtgziAXAbVASwCZJHSAHASQBEQAaEAez2gEMwKpsBCAXjJBjoGMALbKCHAoAbAG4Qs5AOZ0I2AIIA5EgF9S6LIhAYIwiJEmVqUOg2xtynMLyQAGabIXKQKgLrkAVhTQA7BChwwOgBXBHccBjAkYDUQYTQYNCjEAEZbdPJ4xLAAeQAzPLh9OxUgA&indicators=NobwRAlgdgJhDGBDALgewE4EkAiYBcYyEyANgKZgA0YUiAthQVWAM4bL4AMAvpeNHCRosuAi2QoAri2a0G%2BMMzboOeHn0iwEKDDgWJ4RVFABCk5Giiz6jRdWWqeAXSA%3D");
+    await page.getByRole('button', { name: 'Share Report' }).click();
+    await page.getByText('https://host.docker.internal/').click();
+    await page.getByText('This link can be shared to provide a live view into this report.https://host.').click();
+    await page.getByRole('button', { name: 'Close' }).click();
+    //await page.getByRole('button', { name: 'Email Report', exact: true }).click();
+});
+
+test("Test Edit Labels/JSON button working", async ({ page }) => {
+    await page.goto("https://host.docker.internal/Test_Request_Portal/?a=reports&v=3&query=N4IgLgpgTgtgziAXAbVASwCZJHSAHASQBEQAaEAez2gEMwKpsBCAXjJBjoGMALbKCHAoAbAG4Qs5AOZ0I2AIIA5EgF9S6LIhAYIwiJEmVqUOg2xtynMLyQAGabIXKQKgLrkAVhTQA7BChwwOgBXBHccBjAkYDUQYTQYNCjEAEZbdPJ4xLAAeQAzPLh9OxUgA&indicators=NobwRAlgdgJhDGBDALgewE4EkAiYBcYyEyANgKZgA0YUiAthQVWAM4bL4AMAvpeNHCRosuAi2QoAri2a0G%2BMMzboOeHn0iwEKDDgWJ4RVFABCk5Giiz6jRdWWqeAXSA%3D");
+    await page.getByRole('button', { name: 'JSON' }).click();
+    await page.getByRole('button', { name: 'Shorten Link' }).click();
+    await page.getByRole('button', { name: 'Expand Link' }).click();
+    await page.getByRole('button', { name: 'Close' }).click();
+});
+
+test('validate create row button', async ({ page }) => {
```

Comment:
This test should check that the create row button actually works.

Reply:
resolved